### PR TITLE
chore(sg): show cloud ephemeral faq

### PR DESCRIPTION
**Pull request by burmudar**
* show a link to the FAQ on certain errors
* add FAQ command to open the FAQ page

Closes  DINF-150

## Test plan
Locally
```
⚠️ Triggering Cloud Ephemeral builds from "main" is not supported.

  Alternatively, if you still want to deploy "main" you can do:

  1. create a new branch off main by running  git switch <branch-name>
  2. push the branch to the remote by running  git push -u origin <branch-name>
  3. trigger the build by running  sg cloud ephemeral build

  FAQ https://www.notion.so/sourcegraph/How-to-deploy-my-branch-on-an-ephemeral-Cloud-instance-dac45846ca2a4e018c802aba37cf6465?pvs=4#20cb92ae27464891a9d03650b4d67cee
```

### CODE REVIEW

File: dev/sg/internal/cloud/build_command.go [hunk 1]
```diff
@@ -39,7 +39,8 @@ func buildCloudEphemeral(ctx *cli.Context) error {
 		return errors.Wrap(err, "failed to determine current commit")
 	}
 	currRepo := repo.NewGitRepo(currentBranch, head)
-	build, err := triggerEphemeralBuild(ctx.Context, currRepo)
+	err = ErrMainBranchBuild
+	//build, err := triggerEphemeralBuild(ctx.Context, currRepo)
```
Comment (BolajiOlajide) [inline dev/sg/internal/cloud/build_command.go:43]: @burmudar I think you forgot to add this back

**Review by BolajiOlajide — CHANGES_REQUESTED** [review]

File: dev/sg/internal/cloud/build_command.go (hunk 1 above)
Comment (burmudar) [reply dev/sg/internal/cloud/build_command.go:43]: Oh yes 😮‍💨 Thanks!

Comment (burmudar) [reply dev/sg/internal/cloud/build_command.go]: Fixed!

File: dev/sg/internal/cloud/deploy_command.go [hunk 2]
```diff
@@ -229,7 +229,7 @@ Please make sure you have either pushed or pulled the latest changes before tryi
 				steps := "1. create a new branch off main by running `git switch <branch-name>`\n"
 				steps += "2. push the branch to the remote by running `git push -u origin <branch-name>`\n"
 				steps += "3. trigger the build by running `sg cloud ephemeral build`\n"
-				std.Out.WriteMarkdown(fmt.Sprintf("Alternatively, if you still want to deploy \"main\" you can do:\n%s", steps))
+				std.Out.WriteMarkdown(withFAQMarkdown(fmt.Sprintf("Alternatively, if you still want to deploy \"main\" you can do:\n%s", steps)))
```
Comment (burmudar) [inline dev/sg/internal/cloud/deploy_command.go:232]:
```suggestion
				std.Out.WriteMarkdown(withFAQ(fmt.Sprintf("Alternatively, if you still want to deploy \"main\" you can do:\n%s", steps)))
```

**Review by BolajiOlajide — APPROVED** [review]

**Pull request merged**